src/bashi/filter_software_dependency.py

```diff
@@ -103,5 +103,16 @@ def software_dependency_filter(
                     f"{row[CMAKE].version}",
                 )
                 return False
-
+    # Rule: d4
+    # all ROCm images are Ubuntu 20.04 based
```

Comment:
```suggestion
    # Rule: d3
    # all ROCm images are Ubuntu 20.04 based or newer
    # realted to rule c19
```

The next rule number is d3 ;-)

Don't forget to modify the the names of the test functions.